engine/src/main/java/org/terasology/world/block/tiles/WorldAtlasImpl.java

```diff
@@ -275,6 +303,23 @@ private void createTextureAtlas(final Texture texture) {
         return data;
     }
 
+    // Ref: http://stackoverflow.com/questions/221830/set-bufferedimage-alpha-mask-in-java/8058442#8058442
+    public void applyGrayscaleMaskToAlpha(BufferedImage image, BufferedImage mask) {
+        int width = image.getWidth();
+        int height = image.getHeight();
+
+        int[] imagePixels = image.getRGB(0, 0, width, height, null, 0, width);
+        int[] maskPixels = mask.getRGB(0, 0, width, height, null, 0, width);
+
+        for (int i = 0; i < imagePixels.length; i++) {
+            int color = imagePixels[i] & 0x00ffffff; // Mask preexisting alpha
+            int alpha = maskPixels[i] << 24; // Shift blue to alpha
+            imagePixels[i] = color | alpha;
+        }
+
+        image.setRGB(0, 0, width, height, imagePixels, 0, width);
+    }
+
```

Comment:
I don't get this, mostly because I do not know what the Atlas does. Can you please let me know what is being combined here? There's a block that has color texture, normal texture and glossyness texture. What gets combined with what?